src/components/Form/DomainConfiguration.tsx

```diff
@@ -0,0 +1,144 @@
+'use client';
+import React from 'react';
+import { useState } from 'react';
+import { getSubdomain } from '../../modules/MultiDomain/domains';
+import { useDomainStatus } from '../../modules/MultiDomain/UseDomainStatus';
+import { AlertCircle, CheckCircle, XCircle } from 'lucide-react';
+
+export const InlineSnippet = ({ children }: { children: string }) => {
+  return (
+    <span className='inline-block rounded-md bg-blue-100 px-1 py-0.5 font-mono text-blue-900 dark:bg-blue-900 dark:text-blue-100'>
+      {children}
+    </span>
+  );
+};
+export default function DomainConfiguration({ domain }: { domain: string }) {
+  const [recordType, setRecordType] = useState<'A' | 'CNAME'>('A');
+
+  const { status, domainJson } = useDomainStatus({ domain });
+
+  if (!status || status === 'Valid Configuration' || !domainJson) return null;
+
+  const subdomain = getSubdomain(domainJson.name, domainJson.apexName);
+  const txtVerification =
+  (status === 'Pending Verification' &&
+  domainJson.verification.find((x: any) => x.type === 'TXT')) ||
+  null;
+  
+  return (
+    <div className='border-t border-stone-200 px-10 pb-5 pt-7 dark:border-stone-700'>
+      <div className='mb-4 flex items-center space-x-2'>
+        {status === 'Pending Verification' ? (
+          <AlertCircle
+            fill='#FBBF24'
+            stroke='currentColor'
+            className='text-white dark:text-black'
+          />
+        ) : (
+          <XCircle fill='#DC2626' stroke='currentColor' className='text-white dark:text-black' />
+        )}
+        <p className='text-lg font-semibold dark:text-white'>{status}</p>
+        
+      </div>
+      {txtVerification ? (
+        <>
+          <p className='text-sm dark:text-white'>
+            Please set the following TXT record on{' '}
+            <InlineSnippet>{domainJson.apexName}</InlineSnippet> to prove ownership of{' '}
+            <InlineSnippet>{domainJson.name}</InlineSnippet>:
+          </p>
+          <div className='my-5 flex items-start justify-start space-x-10 rounded-md bg-stone-50 p-2 dark:bg-stone-800 dark:text-white'>
+            <div>
+              <p className='text-sm font-bold'>Type</p>
+              <p className='mt-2 font-mono text-sm'>{txtVerification.type}</p>
+            </div>
+            <div>
+              <p className='text-sm font-bold'>Name</p>
+              <p className='mt-2 font-mono text-sm'>
+                {txtVerification.domain.slice(
+                  0,
+                  txtVerification.domain.length - domainJson.apexName.length - 1,
```

Comment:
could be clean to have this small function in an utils file, like that we can easily add some test around